src/preact/base-element.js

```diff
@@ -725,6 +727,21 @@ function collectProps(Ctor, element, ref, defaultProps) {
       devAssert(def.attrs);
       value = def.parseAttrs(element);
     }
+    if (def.attrPrefix) {
+      const currObj = {};
+      const attrs = element.attributes;
+      for (let i = 0; i < attrs.length; i++) {
+        const attrib = attrs[i];
+        if (attrib.name.startsWith(def.attrPrefix)) {
+          currObj[
+            dashToCamelCase(
```

Comment:
The case where `attrib.name === def.attrPrefix` is a real possibility here so we need to:
1. Code for it.
2. Add a test for it.

Off the top of my head I think we should ignore `attrib.name === def.attrPrefix`, since empty key doesn't make sense. Both here and for mutations. But let me know what you think.

Reply:
I agree with you, it should be ignored.